src/libsnark.ml

```diff
@@ -1138,6 +1166,21 @@ struct
       let r = Cpp_string.to_string s in
       Cpp_string.delete s ; Md5.of_binary_exn r
 
+    let evaluations =
+      let evals_struct = structure "r1cs_evalutions" in
+      let a = field evals_struct "a" Field_vector.typ in
+      let b = field evals_struct "b" Field_vector.typ in
+      let c = field evals_struct "c" Field_vector.typ in
+      seal evals_struct ;
+      let stub =
+        foreign
+          (func_name "evaluations")
+          (typ @-> Field_vector.typ @-> size_t @-> returning evals_struct)
+      in
+      fun t ~full_assignment ~degree ->
+        let res = stub t full_assignment (Unsigned.Size_t.of_int degree) in
```

Comment:
Should we assert here that `degree >= num_inputs + num_constraints + 1`? I'm concerned that this may be a source for segfaults/memory corruption later otherwise.